### PR TITLE
Removed some un needed flash messages raised as bugs.

### PR DESCRIPTION
**Pull request by ashimali**
In the process found a couple of edge cases of incorrect use of invitation links by other users which are now handled.

### CODE REVIEW

File: tests/app/main/views/test_accept_invite.py [hunk 1]
```diff
@@ -61,9 +59,9 @@ def test_existing_user_with_no_permissions_accept_invite(app_,
             assert response.status_code == 302
```
Comment (servingUpAces) [inline tests/app/main/views/test_accept_invite.py]: Should we assert the mock_add_user_to_service is being called?

File: tests/app/main/views/test_accept_invite.py [hunk 2]
```diff
@@ -122,15 +119,13 @@ def test_existing_signed_out_user_accept_invite_redirects_to_sign_in(app_,
...
-            mock_accept_invite.assert_called_with(expected_service, sample_invite['id'])
 
             assert response.status_code == 200
             page = BeautifulSoup(response.data.decode('utf-8'), 'html.parser')
             assert page.h1.string.strip() == 'Sign in'
             flash_banners = page.find_all('div', class_='banner-default')
-            assert len(flash_banners) == 2
+            assert len(flash_banners) == 1
             assert flash_banners[0].text.strip() == 'Please log in to access this page.'
-            assert flash_banners[1].text.strip() == 'You already have an account with GOV.UK Notify. Sign in to your account to accept this invitation.'  # noqa
```
Comment (servingUpAces) [inline tests/app/main/views/test_accept_invite.py]: Should we check the mocks are being called appropriately?

**Comment by servingUpAces** [comment]
Let's add some assertions for the mocks in the unit tests, then it should be ready to go.

File: tests/app/main/views/test_accept_invite.py (hunk 1 above)
Comment (ashimali) [inline tests/app/main/views/test_accept_invite.py]: it is :)

**Comment by servingUpAces** [comment]
:-1:
